client/src/components/JurisdictionAdmin/BatchRoundDataEntry.tsx

```diff
@@ -0,0 +1,645 @@
+}
+
+interface IProps {
+  electionId: string
+  jurisdictionId: string
+  roundId: string
+}
+
+const BatchRoundDataEntry: React.FC<IProps> = ({
+  electionId,
+  jurisdictionId,
+  roundId,
+}) => {
+  const batchesResp = useBatches(electionId, jurisdictionId, roundId)
+  const contests = useContestsJurisdictionAdmin(electionId, jurisdictionId)
+  const recordBatchResults = useRecordBatchResults(
+    electionId,
+    jurisdictionId,
+    roundId
+  )
+  const { confirm, confirmProps } = useConfirm()
+
+  const [searchQuery, setSearchQuery] = useState('')
+  const [selectedBatchId, setSelectedBatchId] = useState<IBatch['id']>()
+  const [selectedTabId, setSelectedTabId] = useState('')
+  const [isEditing, setIsEditing] = useState(false)
+  const [isRenamingSheet, setIsRenamingSheet] = useState(false)
+  const [sheetNameBeforeRename, setSheetNameBeforeRename] = useState<{
+    index: number
+    name: string
+  }>()
+
+  const batches = batchesResp.isSuccess ? batchesResp.data.batches : []
+  const filteredBatches = batches.filter(batch =>
+    batch.name.toLowerCase().includes(searchQuery.toLowerCase())
+  )
+  const selectedBatch = batches.find(batch => batch.id === selectedBatchId)
+
+  const initialResultTallySheets = selectedBatch?.resultTallySheets.length
+    ? selectedBatch.resultTallySheets
+    : [constructEmptyResultTallySheet(0)]
+  const formMethods = useForm<IBatchRoundDataEntryFormState>({
+    defaultValues: {
+      resultTallySheets: initialResultTallySheets,
+    },
+  })
+  const {
+    control,
+    formState,
+    getValues,
+    handleSubmit,
+    reset,
+    setValue,
+    watch,
+  } = formMethods
+  const {
+    append: appendSheet,
+    fields: sheets,
+    remove: removeSheet,
+  } = useFieldArray<IBatchResultTallySheet>({
+    control,
+    name: 'resultTallySheets',
+  })
+
+  // Reset state whenever a new batch is selected
+  useEffect(() => {
+    reset({ resultTallySheets: initialResultTallySheets })
+    setSelectedTabId(VOTE_TOTALS_TAB_ID)
+    setIsEditing(false)
+  }, [selectedBatchId])
+
+  // Auto-select first search match
+  const [debouncedSearchQuery] = useDebounce(searchQuery)
+  useEffect(() => {
+    if (
+      debouncedSearchQuery &&
+      filteredBatches.length > 0 &&
+      !formState.isDirty
+    ) {
+      setSelectedBatchId(filteredBatches[0].id)
+    }
+  }, [debouncedSearchQuery])
+
+  if (!batchesResp.isSuccess || !contests) {
+    return null
+  }
+
+  const areResultsFinalized = Boolean(batchesResp.data.resultsFinalizedAt)
+  // Batch comparison audits only support a single contest
+  const [contest] = contests
+
+  // ---------- Handlers (START) ----------
+
+  const selectBatch = (batchId: string) => {
+    if (formState.isDirty) {
+      confirm({
+        title: 'Unsaved Changes',
+        description:
+          'You have unsaved changes. ' +
+          'Are you sure you want to leave this batch without saving changes?',
+        yesButtonLabel: 'Yes, Leave Batch without Saving',
```

Comment:
I think we can standardize this copy with the idea of discarding changes that I mentioned below. Maybe just label these "Discard Changes" and "Cancel"